client/go/outline/client.go

```diff
@@ -41,6 +42,11 @@ func (c *Client) ListenPacket(ctx context.Context) (net.PacketConn, error) {
 	return c.pl.ListenPacket(ctx)
 }
 
+// ClientConfig is used to create the Client.
+type ClientConfig struct {
+	Transport config.ConfigNode
```

Comment:
Ditto: What's the relation between vpn config, client config, transport and reporting/session config?

Reply:
VPN config is used on Linux and it's the config to do the entire tunneling.

Client config is just the part that creates a session with the service and establishes the connection. No routing. Transport is the part of the client that defines the transport-layer protocol. Reporting will go in the client to make it easier.